src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -568,5 +580,17 @@ private void CheckSize(int countToAdd)
                 }
             }
         }
+
+        private struct EncodingPreamblePair
```

Comment:
I'm not convinced this should be a struct.  Since we're only ever constructing a handful of these things, but we're iterating through the array that stores them potentially a lot, seems like it could be better to keep the size of the data being enumerated at one instead of two references.